subsys/bootloader/CMakeLists.txt

```diff
@@ -10,3 +10,12 @@ add_subdirectory_ifdef(CONFIG_SECURE_BOOT_CRYPTO bl_crypto)
 
 include(${CMAKE_CURRENT_LIST_DIR}/cmake/fw_info_magic.cmake)
 zephyr_include_directories(include)
+
+if (NOT IMAGE_NAME)
+  message(WARNING "Building bootloader without associated application. "
+          "Dummy values will be used for partition configuration."
+          "The idiomatic way of including the immutable bootloader is to set "
+          "the configuration 'CONFIG_SECURE_BOOTLOADER=y' in your applications "
+          "'prj.conf'.")
```

Comment:
We need to be more clear in this message whether this is supported or not.

Reply:
It should be possible to suppress this warning when the user does know what he is doing.

Reply:
`To remove this warning, delete ${CMAKE_CURRENT_LINE} from ${CMAKE_CURRENT_FILE}` 

:) 

Like add a kconfig option to build this without printing this warning? I struggle to see the use case when you want to build the bootloader from this location. We want the user to go through the parent image.

Reply:
One alternative is to not support building from samples/bootloader, but to me that breaks too much with the "i can build all the samples" mindset currently adopted.

Reply:
> I struggle to see the use case when you want to build the bootloader from this location. 

Either it's supported or it's not. I'll leave this up to you to decide.

If it's not supported it should be an error. If it's supported you could add a check for a warning suppression variable. No need for Kconfig.

Reply:
It should be supported. I'm not able to find examples of warning suppression variables, do you have any?

Reply:
`if((CMAKE_BUILD_TYPE IN_LIST build_types) AND (NOT NO_BUILD_TYPE_WARNING))
`

Reply:
Always warning is OK